src/sentry/static/sentry/app/views/settings/project/projectServiceHooks.jsx

```diff
@@ -0,0 +1,161 @@
+import ApiMixin from '../../../mixins/apiMixin';
+import AsyncView from '../../asyncView';
+import {BooleanField} from '../../../components/forms';
+import Button from '../../../components/buttons/button';
+import EmptyMessage from '../components/emptyMessage';
+import IndicatorStore from '../../../stores/indicatorStore';
+import {Panel, PanelBody, PanelItem, PanelHeader} from '../../../components/panels';
+import SettingsPageHeader from '../components/settingsPageHeader';
+
+const ServiceHookRow = createReactClass({
+  displayName: 'ServiceHookRow',
+
+  propTypes: {
+    orgId: PropTypes.string.isRequired,
+    projectId: PropTypes.string.isRequired,
+    hook: PropTypes.object.isRequired,
+    onToggleActive: PropTypes.func.isRequired,
+  },
+
+  mixins: [ApiMixin],
+
+  getInitialState() {
+    return {
+      loading: false,
+      error: false,
+    };
+  },
+
+  render() {
+    let {orgId, projectId, hook} = this.props;
+    return (
+      <PanelItem>
+        <Box flex="1">
+          <h5 style={{margin: '10px 0px'}}>
+            <Link to={`/settings/${orgId}/${projectId}/hooks/${hook.id}/`}>
+              {hook.url}
+            </Link>
+          </h5>
+          <small>{hook.events.join(', ')}</small>
+        </Box>
+        <Box width={50}>
+          <BooleanField
+            name={`isActive[${hook.id}]`}
+            onChange={this.props.onToggleActive}
+            value={hook.status == 'active'}
+          />
+        </Box>
+      </PanelItem>
+    );
+  },
+});
+
+export default class ProjectServiceHooks extends AsyncView {
+  static contextTypes = {
+    organization: PropTypes.object.isRequired,
+  };
+
+  getEndpoints() {
+    let {orgId, projectId} = this.props.params;
+    return [['hookList', `/projects/${orgId}/${projectId}/hooks/`]];
+  }
+
+  onToggleActive = hook => {
+    let {orgId, projectId} = this.props.params;
+    let loadingIndicator = IndicatorStore.add(t('Saving changes..'));
+    this.api.request(`/projects/${orgId}/${projectId}/hooks/${hook.id}/`, {
+      method: 'PUT',
+      data: {
+        isActive: hook.status !== 'active',
+      },
+      success: data => {
+        IndicatorStore.remove(loadingIndicator);
+        let hookList = this.state.hooklist.map(h => {
+          if (h.id === data.id) {
+            return {
+              ...h,
+              ...data,
+            };
+          }
+          return h;
+        });
+        this.setState({hookList});
+      },
+      error: () => {
+        IndicatorStore.remove(loadingIndicator);
+        IndicatorStore.add(
+          t('Unable to remove application. Please try again.'),
+          'error',
+          {
+            duration: 3000,
+          }
+        );
+      },
+    });
+  };
+
+  renderEmpty() {
+    return (
+      <EmptyMessage>
+        {t('There are no service hooks associated with this project.')}
+      </EmptyMessage>
+    );
+  }
+
+  renderResults() {
+    let {orgId, projectId} = this.props.params;
+
+    return [
```

Comment:
`<React.Fragment>` instead of array here

Reply:
whats the actual reason for fragment vs array?

Reply:
Functionally they're both the same, but using Fragment is a bit cleaner since the reason we use an array like this previously was as a work-around to avoid having to wrap this in an extra DOM node.

Reply:
If there's no negative to the array though, and React supports that natively (like children), why wouldn't we prefer the array? It's one less component to deal w/

Reply:
imo because it should look more like a component. If for some reason we did need to wrap this with a `<div>`, you wouldn't it as an array.